sharding-proxy/sharding-proxy-backend/src/main/java/org/apache/shardingsphere/shardingproxy/backend/text/sctl/hint/ShardingCTLHintBackendHandler.java

```diff
@@ -0,0 +1,123 @@
+import org.apache.shardingsphere.shardingproxy.backend.response.BackendResponse;
+import org.apache.shardingsphere.shardingproxy.backend.response.error.ErrorResponse;
+import org.apache.shardingsphere.shardingproxy.backend.response.query.QueryData;
+import org.apache.shardingsphere.shardingproxy.backend.response.update.UpdateResponse;
+import org.apache.shardingsphere.shardingproxy.backend.text.TextProtocolBackendHandler;
+import org.apache.shardingsphere.shardingproxy.backend.text.sctl.exception.InvalidShardingCTLFormatException;
+import org.apache.shardingsphere.shardingproxy.backend.text.sctl.exception.UnsupportedShardingCTLTypeException;
+import org.apache.shardingsphere.shardingproxy.backend.text.sctl.hint.internal.HintCommand;
+import org.apache.shardingsphere.shardingproxy.backend.text.sctl.hint.internal.HintType;
+import org.apache.shardingsphere.shardingproxy.backend.text.sctl.hint.internal.command.HintAddDatabaseShardingValueCommand;
+import org.apache.shardingsphere.shardingproxy.backend.text.sctl.hint.internal.command.HintAddTableShardingValueCommand;
+import org.apache.shardingsphere.shardingproxy.backend.text.sctl.hint.internal.command.HintClearCommand;
+import org.apache.shardingsphere.shardingproxy.backend.text.sctl.hint.internal.command.HintSetCommand;
+import org.apache.shardingsphere.shardingproxy.backend.text.sctl.hint.internal.command.HintSetDatabaseShardingValueCommand;
+
+/**
+ * Sharding CTL hint backend handler.
+ *
+ * @author liya
+ */
+public final class ShardingCTLHintBackendHandler implements TextProtocolBackendHandler {
+
+    private static final ThreadLocal<HintManager> HINT_MANAGER_HOLDER = new ThreadLocal<>();
+
+    private final String sql;
+
+    private final boolean supportHint;
+
+    private HintManager hintManager;
+
+    public ShardingCTLHintBackendHandler(final String sql, final BackendConnection backendConnection) {
+        this.sql = sql;
+        this.supportHint = backendConnection.isSupportHint();
+        initHintManagerr(supportHint);
+    }
+
+    private void initHintManagerr(final boolean supportHint) {
+        if (!supportHint) {
+            return;
+        }
+        if (HINT_MANAGER_HOLDER.get() == null) {
+            HINT_MANAGER_HOLDER.set(HintManager.getInstance());
+        }
+        hintManager = HINT_MANAGER_HOLDER.get();
+    }
+
+    @Override
+    public BackendResponse execute() {
+        if (!supportHint) {
+            throw new UnsupportedOperationException(String.format("%s should be true, please check your config", ShardingPropertiesConstant.PROXY_HINT_ENABLED.getKey()));
+        }
+        Optional<ShardingCTLHintStatement> shardingTCLStatement = new ShardingCTLHintParser(sql).doParse();
+        if (!shardingTCLStatement.isPresent()) {
+            return new ErrorResponse(new InvalidShardingCTLFormatException(sql));
+        }
+        HintCommand hintCommand = shardingTCLStatement.get().getHintCommand();
```

Comment:
you can try to use strategy design pattern here to make code more clear.
e.g. add HintCommandExecutor interface, use `HintCommandExecutorFacotry.newInstance` to refactor it.